### PR TITLE
optimize tls example

### LINKED ISSUE #4381: Tls-tidbcluster can't run.
## Bug Report

**What version of Kubernetes are you using?**
1.21.0

**What version of TiDB Operator are you using?**
master

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**
local pv

**What's the status of the TiDB cluster pods?**
basic-discovery-69dfc59d78-gbdrb   1/1     Running            0          21m
basic-pd-0                         0/1     CrashLoopBackOff   8          21m

**What did you do?**
kubectl apply --validate=false -f https://github.com/jetstack/cert-manager/releases/download/v0.13.0/cert-manager.yaml
kubectl apply -f examples/heterogeneous-tls/cert-manager/basic/ 
kubectl apply  -f examples/heterogeneous-tls/tidb-cluster.yaml

**What did you expect to see?**
tidbcluster run.

**What did you see instead?**
image

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
Fix #4381.

### What is changed and how does it work?
Ref: [9398](https://github.com/jetstack/cert-manager/issues/2407)

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Review by csuzhangxc — COMMENTED** [review]
did this only work for some versions for cert-manager?

BTW, I still don't understand `digital signature` and `key encipherment` in https://github.com/cert-manager/cert-manager/issues/2407. 😢

File: examples/heterogeneous-tls/cert-manager/heterogeneous/pd-cluster-secret.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: cert-manager.io/v1alpha2
+apiVersion:  cert-manager.io/v1
```
Comment (DanielZhangQD) [inline examples/heterogeneous-tls/cert-manager/heterogeneous/pd-cluster-secret.yaml:1]:
```suggestion
apiVersion: cert-manager.io/v1
```

File: examples/heterogeneous-tls/cert-manager/heterogeneous/pd-cluster-secret.yaml [hunk 2]
```diff
@@ -7,12 +7,15 @@ spec:
...
-    - PingCAP
+  subject:
+    organizations:
+      - PingCAP
   commonName: "TiDB"
   usages:
     - server auth
     - client auth
+    - digital signature
+    - key encipherment
```
Comment (DanielZhangQD) [inline examples/heterogeneous-tls/cert-manager/heterogeneous/pd-cluster-secret.yaml:18]:
```suggestion
```

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**